opendevin/runtime/client/client.py

```diff
@@ -61,55 +62,102 @@ def __init__(self, plugins_to_load: list[Plugin], work_dir: str) -> None:
         self.shell = pexpect.spawn('/bin/bash', encoding='utf-8', echo=False)
-        self.__bash_PS1 = r'[PEXPECT_BEGIN] \u@\h:\w [PEXPECT_END]'
-
-        # This should NOT match "PS1=\u@\h:\w [PEXPECT]$" when `env` is executed
-        self.__bash_expect_regex = (
-            r'\[PEXPECT_BEGIN\] ([a-z0-9_-]*)@([a-zA-Z0-9.-]*):(.+) \[PEXPECT_END\]'
+        self.prompt_uuid = str(uuid.uuid4())
+        self.__bash_PS1 = (
+            f'[PEXPECT_BEGIN_{self.prompt_uuid}] $ [PEXPECT_END_{self.prompt_uuid}]'
         )
+        print(self.__bash_PS1)
+        self.__bash_expect_regex = f'\\[PEXPECT_BEGIN_{self.prompt_uuid}\\] \\$ \\[PEXPECT_END_{self.prompt_uuid}\\]'
 
         self.shell.sendline(f'export PS1="{self.__bash_PS1}"')
-        self.shell.expect(self.__bash_expect_regex)
+        self._expect_prompt()
+
+        # Disable bracketed paste mode
+        self.shell.sendline("bind 'set enable-bracketed-paste off'")
+        self._expect_prompt()
 
         self.shell.sendline(f'cd {work_dir}')
-        self.shell.expect(self.__bash_expect_regex)
+        self._expect_prompt()
 
     def _get_bash_prompt(self):
-        ps1 = self.shell.after
-        # parse the ps1 to get username, hostname, and working directory
-        matched = re.match(self.__bash_expect_regex, ps1)
-        assert (
-            matched is not None
-        ), f'Failed to parse bash prompt: {ps1}. This should not happen.'
-        username, hostname, working_dir = matched.groups()
-
-        # re-assemble the prompt
-        prompt = f'{username}@{hostname}:{working_dir} '
-        if username == 'root':
-            prompt += '#'
-        else:
-            prompt += '$'
-        return prompt + ' '
-
-    def _execute_bash(self, command, keep_prompt: bool = True) -> tuple[str, int]:
-        logger.debug(f'Executing command: {command}')
+        prompt = self.shell.after
+        # Extract the actual prompt content between the PEXPECT markers
+        match = re.search(r'\[PEXPECT_BEGIN_.*?\] \$ \[PEXPECT_END_.*?\]', prompt)
+        if match:
+            return match.group(0)
+        return prompt
+
+    def _execute_bash(
+        self, command, keep_prompt: bool = True, timeout=30
+    ) -> tuple[str, int]:
+        logger.info(f'Executing command: {command}')
         self.shell.sendline(command)
-        self.shell.expect(self.__bash_expect_regex)
 
-        output = self.shell.before
+        output = ''
+        while True:
+            try:
+                index = self.shell.expect(
+                    [self.__bash_expect_regex, r'\(y/n\)|\[y/N\]'], timeout=timeout
+                )
+                output += self.shell.before
+                if index == 0:  # Matched our prompt
+                    break
+                elif index == 1:  # Matched an interactive prompt
+                    self.shell.sendline('n')  # Default to 'no' for safety
+                    output += "(Automatically responded 'n' to prompt)\n"
```

Comment:
I'm thinking if we can just return an `CmdRunOutputObservation` directly to the agent with this prefix (with the `(y/n)` added there of course): something like 
```
CmdRunOutputObservation(content="XXXX (y/n)", exit_code=None)
# We can't get exit code here bc the shell is abnormal
```

And then we can let agent to decide by either running
```
<execute_bash> y </execute_bash>
OR
<execute_bash> n </execute_bash>
```

Reply:
That'd be awesome, if that would work!